baybe/simulation/scenarios.py

```
@@ -103,34 +168,30 @@ def make_xyzpy_callable(result_variable: str) -> Callable:
         def simulate(
             Scenario: str,
             Monte_Carlo_Run: int,
-            Initial_Data=None,
+            Random_Seed: int,
+            Initial_Data: int | None = None,
         ):
             """Callable for xyzpy simulation."""
             data = None if initial_data is None else initial_data[Initial_Data]
-            seed = Monte_Carlo_Run + (
-                _DEFAULT_SEED if random_seed is None else random_seed
-            )
             result = _simulate_groupby(
                 scenarios[Scenario],
                 lookup,
                 batch_size=batch_size,
                 n_doe_iterations=n_doe_iterations,
                 initial_data=data,
                 groupby=groupby,
-                random_seed=seed,
+                random_seed=Random_Seed,
                 impute_mode=impute_mode,
                 noise_percent=noise_percent,
             )
-            if random_seed is not None:
-                result["Random_Seed"] = seed
             return SimulationResult(result)
 
         return simulate
 
     def unpack_simulation_results(array: DataArray) -> pd.DataFrame:
         """Turn the xyzpy simulation results into a flat dataframe."""
         # Convert to dataframe and remove the wrapper layer
-        series = array.to_series()
+        series = array.to_series().dropna()
```

can you comment why?

---

interesting, I didnt even notice you now call a different function interfacing with xyzpy (just thought you renamed the variable)

thansk for the explanation, seems to make sense, the resutls are always recorded on this exhaustive grid even when instructed to run simulations on a subset of them. THe ones that ar enot in the susbet are not simulated and can be dropped. This seems inefficient at least from a data architecture pov but there must be a reaosn internally or due to features we dont use and I can live with that

my main point is: this is super hard to see from anywhere in our code. Can you add more explanations such that this is understandable without reading half of the xyzpy doc?

---

You mean in form of comments? Let me suggest: I promise that one of my next bigger workitem is the simulation startegy, where I'll also streamline all this I/O stuff and function calls (e.g. fixing the terrible `NothingToSimulate` swallowing of all errors). I could now add detailed explanations but potentially parts of this will become obsolete anyway in the hopefully simplified call stack. So can we postpone cosmetics to that later point to avoid double-work?